src/d3a/d3a_core/simulation.py

```diff
@@ -97,6 +98,17 @@ def __init__(self, setup_module_name: str, simulation_config: SimulationConfig =
         self._init(**self.initial_params)
         self._init_events()
         validate_const_settings_for_simulation()
+        self.count_load_devices_in_setup(self.area, self.load_count)
+        self.endpoint_buffer = SimulationEndpointBuffer(redis_job_id, self.initial_params,
+                                                        self.load_count)
+
+    @classmethod
+    def count_load_devices_in_setup(cls, area, load_count):
```

Comment:
This function could also be part of the MarketUnmatchedLoad class, and to be called in its constructor.